include/dylib.hpp

```diff
@@ -89,7 +89,7 @@ class dylib {
     dylib& operator=(const dylib&) = delete;
 
     dylib(dylib &&other) noexcept : m_handle(other.m_handle) {
-        other.m_handle = nullptr;
+        other.m_handle = 0;
```

Comment:
0 is not really much different than nullptr - it assumes HINSTANCE can be initialized by an integer. You might as well keep the nullptr. If you don't want to do something fancier/more complex, you could at least add a comment regarding the assumption somewhere. Maybe even here.